### ISSUE TITLE
Stopping minishift instance on remote host throws "Error stopping cluster: generic driver does not support stop"

### ISSUE THREAD

**Issue by amitkrout** [minishift/minishift#2458]
### General information

  * Minishift version: master
  * OS: macOS
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Steps to reproduce

  1. minishift profile set foo
  2. export MINISHIFT_ENABLE_EXPERIMENTAL=y
  3. minishift start --no-provision --vm-driver virtualbox
  4. minishift ip
```
192.168.99.101
```
  5. minishift profile set minishift
  6. minishift start --vm-driver generic --remote-ipaddress 192.168.99.101 --remote-ssh-user docker --remote-ssh-key ~/.minishift/profiles/foo/machines/foo/id_rsa
  7. minishift stop
```
$ ./minishift stop
Stopping local OpenShift cluster...
Error stopping cluster: generic driver does not support stop
```

### Expected
stop should successful

### Actual
Reports error ```Error stopping cluster: generic driver does not support stop```

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by gbraad** [comment]
I think this is as expected...

**Comment by praveenkumar** [comment]
@amitkrout this is expected because we don't want to a remote machine to shutdown and `oc cluster` doesn't have any option for stop.

`minishift delete` on the other hand will do `oc cluster down` in case of the generic driver.

**Comment by amitkrout** [comment]
> host to shutdown

I am testing the feature on my local setup through multiple profiles so could not realize the cost of shutting down the host. 

> minishift delete on the other hand will do oc cluster down in case of the generic driver. Note: not as reported, but what about minishift delete ?

Working as expected
```
$ ./minishift status
Minishift:  Running
Profile:    foo
OpenShift:  Stopped
DiskUsage:  8% of 19G
CacheUsage: 884.3 MB (used by oc binary, ISO or cached images)
```

Thanks for the info. Will try some more scenarios on remote setup. 

Closing the issue.

**Comment by gbraad** [comment]
@praveenkumar 

well... wouldn't `m stop` have to do `oc cluster down`... while `m delete` should really remove the `/var/lib/minishift` folder on the host ?

**Comment by amitkrout** [comment]
@praveenkumar  I am not sure on how ```m delete``` works on the remote host, however, @gbraad  question looks valid to me

**Comment by agajdosi** [comment]
> well... wouldn't m stop have to do oc cluster down... while m delete should really remove the /var/lib/minishift folder on the host ?

Big :+1: I like this proposal. It is quite understandable behaviour IMHO. WDYT @praveenkumar?

**Comment by praveenkumar** [comment]
Sure that makes sense I will put a PR for this.

**Comment by LalatenduMohanty** [comment]
Fixed by #2479
